### DISCUSSION TITLE
Deploy with ML/DL model could not make all cpus work on Centos7 but win10

### DISCUSSION THREAD

**Discussion by w-Bro** [fastapi/fastapi#6434] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import xxx

ocr_system = OCRSystem(config=settings.OCR_CONFIG_JSON)


@router.post("/ocr")
def ocr(ocr_data: ocr_schema.OcrRequest):
    image = base64_to_cv2(ocr_data.image)
    results = ocr_system.ocr(image, cls=True)
    final_results = []
    for res in results:
        final_results.append({
            # text region
            "textRegion": res[0],
            # text
            "text": res[1][0],
            # confidence
            "confidence": round(res[1][1], 2),
        })
    return response_schema.resp_200(data=final_results)
```

### 2. on win10, my cpu is AMD Ryzen 7 5800H with 8 cores and 16 threads, i found that when `/ocr` is called, all 16 threads work
<img width="291" alt="89d4523d292407352039ef2d0dd13f1" src="https://user-images.githubusercontent.com/24600584/140597612-38b926de-d198-4f9b-a4c0-8a7193e9ac43.png">

### 3. on centos, my cpu is Intel(R) Xeon(R) CPU E5-2680 v4 with 4 cores(this a virtual server, it is cut from a real server with 2 E5-2680 v4 cpus), i found that when `/ocr` is called, usage of cpu is 200%,  it seems means that there are 2 cores working but not all 4 cores.
![image](https://user-images.githubusercontent.com/24600584/140597853-87fc6420-87db-44db-adc7-8bd4cb7c9b1c.png)

anybody know about this and maybe i  am work it with a wrong way?

### Operating System

Linux, Windows

### FastAPI Version

0.68.1

### Python Version

Python 3.7.10

**Comment by raphaelauv** [comment]
As you said this is not a fastAPI issue.

Multiprocessing is a complex topic , there is a lot of different problems and solutions.

Maybe that can be a lead to your problems ->

https://github.com/tiangolo/fastapi/issues/2425#issuecomment-734790381

You can very probably close this issue, thank you.
